src/Utilities/MemoryHelpers.hpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <type_traits>
+
+namespace cpp20 {
+namespace detail {
+template <typename T>
+struct unique_type {
+  using single_object = std::unique_ptr<T>;
+};
+
+template <typename T>
+struct unique_type<T[]> {
+  using array = std::unique_ptr<T[]>;
+};
+
+template<typename T, size_t Bound>
+struct unique_type<T[Bound]> { struct invalid_type { }; };
+}  // namespace detail
+
+template <typename T, typename... Args>
+typename detail::unique_type<T>::single_object make_unique_default_init() {
```

Comment:
I tried to find this on cpp reference and it looks like this was rename to `make_unique_for_overwrite` (https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique). I'm happy to leave this as an issue to do a find-replace on in the future in the interest of getting this merged sooner :)

Reply:
Running sed is easy and I had other changes to make anyway.